docs/architecture/Data_caching_lineage.md

```diff
@@ -0,0 +1,297 @@
+
+This strategy is particularly effective in training computer vision models because large amount of image data can be stored cost effectively in Azure Blob Storage, and archieved in cool tiers over time. Meanwhile, prior to running Azure Machine Learning pipelines, data can be copied to Premium storage for fast access by the training cluster.
+
+### What to cache
+
+When identifying files to be copied between storages for caching, we usually deal with one of two cases:
+
+1. Whole dataset transfer from the long-term storage location to the cache location.
+1. Partial transfer based on a list of files to copy.
+
+Second case is a situation when an actual set of file to use is a significantly smaller part of the dataset. It's very typical for specific experiment pipelines or integration tests. While it may optimize premium storage utilization and caching time, maintaining lists of files to copy requires additional investment, such as annotations processing. For example, if annotations are stored in Pascal VOC format, there may be a code component that builds a list of labeled images so that only these files are cached.
+
+### Recommended Data Caching Strategy
+
+If images are mostly shared between pipelines and experiments in a single cache storage, caching the whole dataset is an optimal choice as it doesn't bring additional complication to data management.
+
+## Data Lineage in Azure Machine Learning
+
+**Model reproducibility** is a critical component of production-grade machine learning pipelines. Without it, Data scientists risk having limited visibility into what causes changes in model performance. The variability may appear to be caused by an adjustment of one parameter, but may actually be caused by hidden sources of randomness. Reproducibility reduces or eliminates variations when rerunning failed jobs or prior experiments, making it essential in the context of fault tolerance and iterative refinement of models. Running ML pipelines in the cloud across multiple compute nodes not only multiplies the sources of non-determinism but also increases the need for both fault tolerance and iterative model development.
+
+There are 5 major components that define a machine learning model:
+
+1. Model architecture
+1. Model hyperparameters
+1. Training code
+1. Data augmentation code and parameters
+1. Data and annotations
+
+First 4 components are tracked between pipeline's Git repository and Azure Machine Learning tracking storages. Therefore, it's relatively easy to achieve model reproducibility up to the point of what data was used.
```

Comment:
I might mention here something parenthetical like (as long as sources of randomness are parameterized as well)